src/wallet/sqlite.cpp

```diff
@@ -273,35 +278,40 @@ void SQLiteDatabase::Open()
     }
 
     if (sqlite3_db_readonly(m_db, "main") != 0) {
-        throw std::runtime_error("SQLiteDatabase: Database opened in readonly mode but read-write permissions are needed");
+        if (!m_read_only) {
```

Comment:
In f6f1f4199b4a3c2d7ef7c1bcfa122a9d8c5eeb64 "wallet: Allow read-only database access for info and dump commands"

The `if`s can be combined:

```suggestion
    if (!m_read_only && sqlite3_db_readonly(m_db, "main") != 0) {
```